sdk/containerregistry/container-registry/src/containerRepository.ts

```diff
@@ -0,0 +1,346 @@
+   */
+  public async getProperties(
+    options: GetRepositoryPropertiesOptions = {}
+  ): Promise<RepositoryProperties> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-getProperties", options);
+
+    try {
+      const result = await this.client.containerRegistry.getProperties(this.name, updatedOptions);
+      return result;
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Updates repository attributes.
+   * @param options -
+   */
+  public async setProperties(
+    options: SetRepositoryPropertiesOptions
+  ): Promise<RepositoryProperties> {
+    const { span, updatedOptions } = createSpan("ContainerRepository-setProperties", {
+      ...options,
+      value: {
+        canDelete: options.canDelete,
+        canWrite: options.canWrite,
+        canList: options.canList,
+        canRead: options.canRead
+      }
+    });
+
+    try {
+      const properties = await this.client.containerRegistry.setProperties(
+        this.name,
+        updatedOptions
+      );
+      return {
+        ...properties,
+        writeableProperties: {
+          canDelete: properties.writeableProperties.canDelete,
+          canList: properties.writeableProperties.canList,
+          canRead: properties.writeableProperties.canRead,
+          canWrite: properties.writeableProperties.canWrite
+        }
+      };
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Iterates manifests.
+   *
+   * Example usage:
+   * ```ts
+   * const client = new ContainerRegistryClient(url, credentials);
+   * const repository = client.getRepository(repositoryName)
+   * for await (const manifest of client.listManifests()) {
+   *   console.log("manifest: ", manifest);
+   * }
+   * ```
+   * @param options -
+   */
+  public listManifests(
+    options: ListManifestsOptions = {}
+  ): PagedAsyncIterableIterator<ArtifactManifestProperties> {
+    const iter = this.listManifestsItems(options);
+
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) => this.listManifestsPage(settings, options)
+    };
+  }
+
+  private async *listManifestsItems(
+    options: ListManifestsOptions = {}
+  ): AsyncIterableIterator<ArtifactManifestProperties> {
+    for await (const page of this.listManifestsPage({}, options)) {
+      yield* page;
+    }
+  }
+
+  private async *listManifestsPage(
+    continuationState: PageSettings,
+    options: ListManifestsOptions = {}
+  ): AsyncIterableIterator<ArtifactManifestProperties[]> {
+    const orderby =
+      options.orderBy === "timeAsc"
+        ? "timeasc"
+        : options.orderBy === "timeDesc"
+        ? "timedesc"
+        : undefined;
```

Comment:
I wonder if `toLowerCase` would be better here. It's not really that relevant in itself, however: These constant values, would they change later? should we scope these transformations in some functions? so that the transformation changes happen separately from the logic changes?

Take it or leave it, I'm just sharing an idea 🌞